src/components/tl/mlx5/mcast/tl_mlx5_mcast_coll.c

```diff
@@ -284,8 +553,9 @@ void ucc_tl_mlx5_mcast_collective_progress(ucc_coll_task_t *coll_task)
     ucc_tl_mlx5_task_t           *task   = ucc_derived_of(coll_task, ucc_tl_mlx5_task_t);
     ucc_tl_mlx5_mcast_coll_req_t *req    = task->coll_mcast.req_handle;
 
-    if (req != NULL) {
-        coll_task->status = ucc_tl_mlx5_mcast_test(req);
+    coll_task->status = (req->progress)(req);
```

Comment:
so `req` cannot be null here?

Reply:
No it won't be, but I will add an assertion just in case